addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
@rwjblue thanks for the TL;DR

My deeper question is - how would this work in the following scenario (run-loop details)

1) user clicks a button w/ onclick
2) that click event is wrapped w/ ember.run (closure action or something lets say)
3) that click event fires this code above
4) when the resolve fires you will most likely not be in that original ember.run (from step 1/2 above) -right?

if you are not as I showed above (total guess -no proof here in code as you can tell) -what is a real use case in which join is successful / useful ?

btw -not trolling here if it sounds that way - just learning a lot about ember.run /when it's used correctly and wanted to learn more :)